sphinx/builders/__init__.py

```diff
@@ -41,7 +41,7 @@
 from sphinx import roles  # NoQA: F401  isort:skip
 
 if TYPE_CHECKING:
-    from collections.abc import Iterable, Sequence
+    from collections.abc import Iterable, Sequence, Set
```

Comment:
I'd suggest using `Set as AbstractSet` (I think that's that the docs and linters usually recommend)